webui/module/Restore/src/Restore/Form/RestoreForm.php

```diff
@@ -32,225 +32,91 @@ class RestoreForm extends Form
+                'value' => $restore_params['client']
+            )
+        ));
 
         // Restore client
-        if (isset($this->restore_params['restoreclient'])) {
-            $this->add(array(
-                'name' => 'restoreclient',
-                'type' => 'select',
-                'options' => array(
-                    'label' => _('Restore to client'),
-                    'empty_option' => _('Please choose a client'),
-                    'value_options' => $this->getClientList()
-                ),
-                'attributes' => array(
-                    'class' => 'form-control selectpicker show-tick',
-                    'data-live-search' => 'true',
-                    'id' => 'restoreclient',
-                    'value' => $this->restore_params['restoreclient']
-                )
-            ));
-        } elseif (!isset($this->restore_params['restoreclient']) && isset($this->restore_params['client'])) {
-            $this->add(array(
-                'name' => 'restoreclient',
-                'type' => 'select',
-                'options' => array(
-                    'label' => _('Restore to client'),
-                    'empty_option' => _('Please choose a client'),
-                    'value_options' => $this->getClientList()
-                ),
-                'attributes' => array(
-                    'class' => 'form-control selectpicker show-tick',
-                    'data-live-search' => 'true',
-                    'id' => 'restoreclient',
-                    'value' => $this->restore_params['client']
-                )
-            ));
-        } else {
-            $this->add(array(
-                'name' => 'restoreclient',
-                'type' => 'select',
-                'options' => array(
-                    'label' => _('Restore to (another) client'),
-                    'empty_option' => _('Please choose a client'),
-                    'value_options' => $this->getClientList()
-                ),
-                'attributes' => array(
-                    'class' => 'form-control selectpicker show-tick',
-                    'data-live-search' => 'true',
-                    'id' => 'restoreclient',
-                    'disabled' => true
-                )
-            ));
-        }
+        $this->add(array(
+            'name' => 'restoreclient',
+            'type' => 'select',
+            'options' => array(
+                'label' => _('Restore to client'),
+                'empty_option' => _('Please choose a client'),
+                'value_options' => $this->restore_target_clients
+            ),
+            'attributes' => array(
+                'class' => 'form-control selectpicker show-tick',
+                'data-live-search' => 'true',
+                'id' => 'restoreclient',
+                'value' => $this->restore_params['restoreclient'] ?? $this->restore_params['client'],
+                'disabled' => (!isset($this->restore_params['client']))
```

Comment:
Do we not already assume that this is set when setting the client above ?

Reply:
(Also applies to all the checks found below)

Reply:
Not sure what you mean. If you refer to "value": if `$restore_params['client']` is not set, `"value"` will be `Null`, which is fine.

Reply:
I think i made the wrong assumption that accessing a non existing key in an array throws an exception, but testing it in the interactive shell shows that it "only" prints a warning, so i guess there is nothing to do here!